### PR TITLE
DDD-16 initial react setup

### PR DESCRIPTION
**Pull request by Accir**
create-react-app with removed dependencies

### CODE REVIEW

File: src/main/webapp/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (Gintasp) [inline src/main/webapp/.gitignore:1]: This file does not belong here. We should move contents into .gitignore at the root of repo.

File: src/main/webapp/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
Comment (Gintasp) [inline src/main/webapp/README.md:1]: Can we merge this with README at the root of repo?

File: src/main/webapp/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (Gintasp) [inline src/main/webapp/package.json:1]: This file should be at the root of repo

Comment (Accir) [reply src/main/webapp/package.json:1]: Will react see package.json file if it's at the root?

**Comment by giplonis** [comment]
I thought for a bit - we do not even need to keep these react files inside java project's webroot since we'll run React separately. I moved them outside java project into `ui` directory. Life will be much easier this way 😄 👍  Btw, I recommend using `yarn` instead of `npm` for development.

File: ui/package.json [hunk 4]
```diff
@@ -1,15 +1,20 @@
 {
-  "name": "webapp",
+  "name": "ui",
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@testing-library/jest-dom": "^5.11.4",
+    "@testing-library/react": "^11.1.0",
+    "@testing-library/user-event": "^12.1.10",
```
Comment (Accir) [inline ui/package.json:8]: Do we need react testing dependencies?

Comment (giplonis) [reply ui/package.json:8]: At least for now - we don't. But I thought there's no harm in keeping them if we decide to write some ui tests (unlikely). If you think it's redundant and it's easy to remove - just do it 🙂

**Comment by Accir** [comment]
Everything looks good, you will need to approve it 😄

**Review by giplonis — APPROVED** [review]

**Pull request merged**